src/realm/geospatial.hpp

```diff
@@ -217,23 +210,8 @@ class Geospatial {
 
     constexpr static std::string_view c_geo_point_type_col_name = "type";
     constexpr static std::string_view c_geo_point_coords_col_name = "coordinates";
-    constexpr static std::string_view c_types[] = {"Point", "Box", "Polygon", "CenterSphere"};
-
 private:
-    Type m_type;
-    std::vector<GeoPoint> m_points;
-
-    double m_radius_radians = get_nan();
-
-    constexpr static double get_nan()
-    {
-        return std::numeric_limits<double>::quiet_NaN();
-    }
-
-    bool is_radius_valid() const
-    {
-        return !std::isnan(m_radius_radians);
-    }
+    mpark::variant<mpark::monostate, GeoPoint, GeoBox, GeoPolygon, GeoCenterSphere> m_value;
```

Comment:
The code looks nicer with this, no doubt, but this increases the size of Geospatial from 56 to 72. Are we ok with this?

Reply:
Now that `Geospatial` does not affect the size of `Mixed` I don't think it matters too much anymore.

Reply:
In essence it looks to me as the same concern as with Mixed, that it increases in-memory footprint of constructed geospatials. But you are right, should be a problem for now. Let's from use cases later if there is something to optimize.